storage/myisam/mi_check.c

```diff
@@ -874,7 +874,7 @@ static int chk_index(HA_CHECK *param, MI_INFO *info, MI_KEYDEF *keyinfo,
   if (keypos != endpos)
   {
     mi_check_print_error(param,"Keyblock size at page %s is not correct.  Block length: %d  key length: %d",
-                llstr(page,llbuff), used_length, (keypos - buff));
+                llstr(page,llbuff), used_length, (int) (keypos - buff));
```

Comment:
Doïng this to match the surrounding (bad) code:
https://github.com/MariaDB/server/blob/cf08f44d031280944af42a44eb52121c45511d09/storage/myisam/mi_check.c#L847-L869

Reply:
@ParadoxV5 you might as well remove `llstr` here completely and replace
```c
   char llbuf[22];
   my_printf("%s", llstr(val, llbuf));
```
with
```c
   my_printf("%llu", val);
```
`llstr()` is a workaround for printing longlongs from the times when `%llu` was not supported

Reply:
> Should the block length and key length use `%u` rather than `%d`?

Probably. `%d` was the existing format; I inferred from it and didn’t question if it’s unintended.